src/dag-node/dagNode.js

```diff
@@ -68,9 +69,19 @@ class DAGNode {
     return toDAGLink(this, options)
   }
 
+  serialize () {
+    return serializeDAGNode({
+      Data: this._data,
+      Links: this._links
+    })
+  }
+
   get size () {
-    if (this._size === undefined) {
-      this._size = this._links.reduce((sum, l) => sum + l.Tsize, this._serializedSize)
+    if (this._size === null) {
+      if (this._serializedSize === null) {
+        this._serializedSize = this.serialize().length
+      }
+      this._size = this._links.reduce((sum, l) => sum + l.Tsize, this.serialize().length)
```

Comment:
`this.serialize().length` is called whenever `this._size === null`. This means that we could just unconditionally assign it to `this._serializedSize`:

```js
get size() {
  if (this._size === null) {
    this._serializedSize = this.serialize().length
    this._size = this._links.reduce((sum, l) => sum + l.Tsize, this._serializedSize)
  }
}
```

Reply:
oh, that's a mistake, changed it to have only that one call to `this.serialize().length`:

```
    if (this._size === null) {
      if (this._serializedSize === null) {
        this._serializedSize = this.serialize().length
      }
      this._size = this._links.reduce((sum, l) => sum + l.Tsize, this._serializedSize)
    }
```

If `this._serializedSize` comes in from the constructor then we only do the outer condition, if it's not in the constructor then we do both. The use in `util.deserialize()` to send `buffer.length` (unlike my mistaken `buffer`, see below) in is still a good optimisation I think and saves a re-serialisation for this size calculation.

Mind you, I have no idea how much this matters, how often is `size()` calculated on real data? How much does it really matter? I just don't have insight to how this is used upstream.

Reply:
I also have no idea. It just seemed easy enough to have it called only once.